nixos/doc/manual/md-to-db.sh

```diff
@@ -0,0 +1,32 @@
+#! /usr/bin/env nix-shell
+#! nix-shell -I nixpkgs=channel:nixpkgs-unstable -i bash -p pandoc
+
+# This script is temporarily needed while we transition the manual to
+# CommonMark. It converts the .md files in the regular manual folder
+# into DocBook files in the from_md folder.
```

Comment:
I don't understand how this is used. This doesn't seem to be called anywhere in the build process.

Do we really want to keep the same manual pages in both docbook and md around in the repo at the same time, and ask everyone doing changes to manually keep them in sync?

Can't we consume `.md` files where things are already converted, and `.xml` where not, like in `doc/languages-frameworks`?

Reply:
The context is that Eelco [does not like Pandoc being in NixOS closure](https://github.com/NixOS/nixpkgs/pull/106847#issuecomment-744503875). But it has been in the closure for a long time now and I do not see much benefit trying to get rid of it during the migration period. Especially when the cost would be keeping artefacts in the repo.

Reply:
Keeping both files in sync won't work. People are not aware they need to edit both files when updating docs, and these files will just diverge.

Yes, pandoc is a heavy ask, but we also build half of [firefox due to polkit rules being in javascript](https://twitter.com/zimbatm/status/1341365282888146950), so having pandoc in the build closure is a small problem - and if people want to quickly iterate on a new rebuild, disabling the manual for that local test is probably the better way forward than dumping two different and slowly diverging documentation file formats.

Reply:
We could have tooling for that – I think adisbladis mentioned they use GitHub actions for that in one of their projects. But seriously, pandoc is less than 200 MB static binary and even the build closure (if someone cannot use the cached version) is [comparable to whatever will replace it](https://github.com/NixOS/nixpkgs/pull/106847#issuecomment-744475018). So removing it only makes sense if we intend to keep this dual arrangement after the migration as well.

Reply:
I'll update this to add myself to CODEOWNERS so I can monitor/enforce proper changes to the files.

Reply:
Oh, wait, so pandoc is already in the closure? Then I don't think we should do it this way.

Reply:
> but we also build half of firefox due to polkit rules being in javascript, so having pandoc in the build closure is a small problem

Bloat in one area shouldn't be an excuse for adding more bloat.

I'm strongly :-1: (as in "veto") on having a GHC dependency in the NixOS base system, even if it's just a build-time dependency.

Reply:
@jtojnar What is currently adding pandoc to the NixOS closure?

Reply:
Oh, I though this is for Nixpkgs manual. NixOS currently does not depend on Pandoc.

Reply:
What's the plan on rendering these NixOS docs in the base system after the migration to markdown is done? What's the closure size of that?

Reply:
> What's the plan on rendering these NixOS docs in the base system after the migration to markdown is done? What's the closure size of that?

We do not have a plan.

Reply:
> I'll update this to add myself to CODEOWNERS so I can monitor/enforce proper changes to the files.

I've added myself to CODEOWNERS

Reply:
> > What's the plan on rendering these NixOS docs in the base system after the migration to markdown is done? What's the closure size of that?
> 
> We do not have a plan.

Then we should probably re-think on this.
If there's no way to produce the manual on NixOS without getting pandoc into the (build) closure. we either need to live with that, or not have any docs in the NixOS base system (which also doesn't sound very good to me either).

@edolstra, what are your thoughts on this?

Reply:
Let's do that discussion in https://github.com/NixOS/nixpkgs/issues/63513 I think.